File: app/src/main/java/fr/free/nrw/commons/upload/PendingUploadsContract.java
package fr.free.nrw.commons.upload;

import android.content.Context;
import fr.free.nrw.commons.BasePresenter;
import fr.free.nrw.commons.contributions.Contribution;
import fr.free.nrw.commons.nearby.contract.NearbyParentFragmentContract;
import fr.free.nrw.commons.nearby.contract.NearbyParentFragmentContract.View;

/**
 * The contract using which the PendingUploadsFragment or FailedUploadsFragment would communicate
 * with its PendingUploadsPresenter
 */
public class PendingUploadsContract {

    /**
     * Interface representing the view for uploads.
     */
    public interface View { }

    /**
     * Interface representing the user actions related to uploads.
     */
    public interface UserActionListener extends
        BasePresenter<fr.free.nrw.commons.upload.PendingUploadsContract.View> {

        /**
         * Deletes a upload.
         */
        void deleteUpload(Contribution contribution, Context context);
    }
}

File: app/src/main/java/fr/free/nrw/commons/upload/PendingUploadsContract.kt
package fr.free.nrw.commons.upload

import android.content.Context
import fr.free.nrw.commons.BasePresenter
import fr.free.nrw.commons.contributions.Contribution

/**
 * The contract using which the PendingUploadsFragment or FailedUploadsFragment would communicate
 * with its PendingUploadsPresenter
 */
class PendingUploadsContract {
    /**
     * Interface representing the view for uploads.
     */
    interface View

    /**
     * Interface representing the user actions related to uploads.
     */
    interface UserActionListener : BasePresenter<View> {
        /**
         * Deletes a upload.
         */
        fun deleteUpload(contribution: Contribution?, context: Context?)
    }
}
